exporter/trace/stackdriver/stackdriver.go

```diff
@@ -0,0 +1,289 @@
+	return func(o *options) {
+		o.OnError = onError
+	}
+}
+
+// WithTraceClientOptions sets additionial client options for tracing.
+func WithTraceClientOptions(opts []option.ClientOption) func(o *options) {
+	return func(o *options) {
+		o.TraceClientOptions = opts
+	}
+}
+
+// WithContext sets the context that trace exporter and metric exporter
+// relies on.
+func WithContext(ctx context.Context) func(o *options) {
+	return func(o *options) {
+		o.Context = ctx
+	}
+}
+
+func (o *options) handleError(err error) {
+	if o.OnError != nil {
+		o.OnError(err)
+		return
+	}
+	log.Printf("Failed to export to Stackdriver: %v", err)
+}
+
+const defaultTimeout = 5 * time.Second
+
+// Exporter is a trace exporter that uploads data to Stackdriver.
+//
+// TODO(yoshifumi): add a metrics exporter once the spec definition
+// process and the sampler implementation are done.
+type Exporter struct {
+	traceExporter *traceExporter
+}
+
+// NewExporter creates a new Exporter thats implements trace.Exporter.
+//
+// TODO(yoshifumi): add a metrics exporter one the spec definition
+// process and the sampler implementation are done.
+func NewExporter(opts ...Option) (*Exporter, error) {
+	o := options{}
+	for _, opt := range opts {
+		opt(&o)
+	}
+	if o.ProjectID == "" {
+		ctx := o.Context
+		if ctx == nil {
+			ctx = context.Background()
+		}
+		creds, err := google.FindDefaultCredentials(ctx, traceapi.DefaultAuthScopes()...)
+		if err != nil {
+			return nil, fmt.Errorf("Stackdriver: %v", err)
+		}
+		if creds.ProjectID == "" {
+			return nil, errors.New("Stackdriver: no project found with application default credentials")
+		}
+		o.ProjectID = creds.ProjectID
+	}
+	te, err := newTraceExporter(&o)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Exporter{
+		traceExporter: te,
+	}, nil
+}
+
+func newContextWithTimeout(ctx context.Context, timeout time.Duration) (context.Context, func()) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	if timeout <= 0 {
+		timeout = defaultTimeout
+	}
+	return context.WithTimeout(ctx, timeout)
+}
+
+var spanProcessor trace.SpanProcessor
+
+// RegisterBatchSpanProcessor registers e as BatchSpanProcessor.
+func (e *Exporter) RegisterBatchSpanProcessor() error {
+	delayThreshold := 2 * time.Second
+	if e.traceExporter.o.BatchDelayThreshold > 0 {
+		delayThreshold = e.traceExporter.o.BatchDelayThreshold
+	}
+	queueSizeThreshold := 50
+	if e.traceExporter.o.QueueSizeThreshold > 0 {
+		queueSizeThreshold = e.traceExporter.o.QueueSizeThreshold
+	}
+
+	spanProcessor, err := trace.NewBatchSpanProcessor(e,
+		trace.WithMaxExportBatchSize(queueSizeThreshold),
+		trace.WithMaxQueueSize(queueSizeThreshold),
+		trace.WithScheduleDelayMillis(delayThreshold),
+	)
+	if err != nil {
+		return err
+	}
+	trace.RegisterSpanProcessor(spanProcessor)
+	return nil
+}
+
+// RegisterSimpleSpanProcessor registers e as SimpleSpanProcessor.
+func (e *Exporter) RegisterSimpleSpanProcessor() {
+	spanProcessor := trace.NewSimpleSpanProcessor(e)
```

Comment:
this creates a new local spanProcessor object and does not update the global one. Anyway, once you move spanProcess inside Exporter you don'e need the global one.

Reply:
Done with the fix that move spanProcessor to Exporter field.